mne/channels/tests/test_montage.py

```diff
@@ -373,6 +375,76 @@ def test_read_locs():
     assert_allclose(pos[:4], expected, atol=1e-7)
 
 
+def _fix_old_polhemus_montage(montage):
+    """Hack to fix the broken historical way."""
+    montage.dig[0]['kind'] = FIFF.FIFFV_POINT_CARDINAL
+    montage.dig[1]['kind'] = FIFF.FIFFV_POINT_CARDINAL
+    montage.dig[2]['kind'] = FIFF.FIFFV_POINT_CARDINAL
+
+    montage.dig[0]['ident'] = FIFF.FIFFV_POINT_NASION
+    montage.dig[1]['ident'] = FIFF.FIFFV_POINT_LPA
+    montage.dig[2]['ident'] = FIFF.FIFFV_POINT_RPA
+
+    montage.dig[0], montage.dig[1] = montage.dig[1], montage.dig[0]
+
+    for k in range(3, 8):
+        montage.dig[k]['ident'] = k - 2
+
+
+def test_read_dig_polhemus():
+    """Test read_dig_polhemus."""
+    # Test old way matches new way
+
+    names = ['nasion', 'lpa', 'rpa', '1', '2', '3', '4', '5']
+    names_new = \
+        ['nasion', 'lpa', 'rpa', 'HPI 1', 'HPI 2', 'HPI 3', 'HPI 4', 'HPI 5']
+    # montage = read_dig_montage(hsp, hpi, elp, names, transform=False)
+    montage = read_dig_montage(hsp=None, hpi=hpi, elp=elp,
+                               point_names=names, transform=False)
+    # montage_polhemus = read_dig_polhemus(hsp=hsp, hpi=hpi, elp=elp,
+    montage_polhemus = read_dig_polhemus(hsp=None, hpi_dev=hpi, elp=elp,
+                                         point_names=names_new)
+
+    # Check the HPI in device coordinate are present
+    dig_hpi_dev = montage_polhemus.dig[-5:]
+    for d in dig_hpi_dev:
+        assert d['kind'] == FIFF.FIFFV_POINT_HPI
+        assert d['coord_frame'] == FIFF.FIFFV_COORD_DEVICE
+
+    _fix_old_polhemus_montage(montage)
+
+    assert montage.dig == montage_polhemus.dig[:-5]
+    assert object_diff(montage.ch_names, montage_polhemus.ch_names) == ''
+
+    # montage = read_dig_montage(hsp, hpi, elp, names, transform=True)
+    montage = read_dig_montage(None, hpi, elp, names, transform=True)
+    _fix_old_polhemus_montage(montage)
+    montage_polhemus = transform_to_head(montage_polhemus)
+    assert montage.dig == montage_polhemus.dig[:-5]
+    assert object_diff(montage.ch_names, montage_polhemus.ch_names) == ''
+
+
+def test_read_dig_polhemus_isotrack():
+    """Test reading polhemus isotrack files."""
+    # Make sure new reader produces the same as old `read_dig_montage`
+    # XXX: `read_dig_montage` was not tested against
+    #      `mne/io/kit/tests/data/test.hsp` (or `test.elp`) this was only
+    #      tested in through the kit reader in (test_kit.py::test_hsp_elp).
+
+    foo = read_dig_montage(hsp=op.join(kit_dir, 'test.hsp'), transform=False)
+    bar = read_dig_montage(elp=op.join(kit_dir, 'test.elp'), transform=False)
+
+    # XXX: check whatever is currently done
+    assert foo.__repr__() == (
+        '<DigMontage | '
+        '500 extras (headshape), 0 HPIs, 0 fiducials, 0 channels>'
+    )
+    assert bar.__repr__() == (
+        '<DigMontage | '
+        '0 extras (headshape), 8 HPIs, 0 fiducials, 0 channels>'
```

Comment:
IMHO `read_dig_montage` is also broken for `elp` files. The [test.elp file](https://github.com/mne-tools/mne-python/blob/751347db894da259c6c2ecaf65e81ab8c4aaffdf/mne/io/kit/tests/data/test.elp) contains 3 fiducials and 5 HPI points properly label (0 to 4). 

see fiducials:
https://github.com/mne-tools/mne-python/blob/751347db894da259c6c2ecaf65e81ab8c4aaffdf/mne/io/kit/tests/data/test.elp#L9-L12

HPI numbering example:
https://github.com/mne-tools/mne-python/blob/751347db894da259c6c2ecaf65e81ab8c4aaffdf/mne/io/kit/tests/data/test.elp#L35-L37

Therefore I would expect `bar.__repr__()` to give something like:
```
<DigMontage | 0 extras (headshape), 5 HPIs, 3 fiducials, 0 channels>'
```